src/env.d.ts

```diff
@@ -9,6 +9,7 @@ interface ImportMetaEnv {
   readonly PUBLIC_INITIAL_DELIMITER: string;
   readonly PUBLIC_PROTECTED_ID_KEY: string;
   readonly PUBLIC_ALLOW_SET_THEME: string;
+  readonly PUBLIC_OKTA_SSO_PATH: string | undefined;
```

Comment:
Oops sorry, there was one thing I missed when reviewing @ttoomey. We don't want to specify whether or not to use okta at build time, we want to specify it at runtime. The frontend should be built once and able to deploy to multiple environments. These variables listed here are installation-wide. (APP_NAME, WAREHOUSE_URL, and CAS_URL all need to be removed, but I haven't gotten to it yet).

We don't support injecting runtime variables into the frontend. Instead, our approach for "configurations" is to have the frontend fetch them on page load and store them in local storage. This is currently how we are fetching themes (`MergedThemeProvider` fetches the theme from `ThemeController`). 

We should be doing something similar for Okta. There are going to be other settings and feature flags, too. To accomplish this, I think we need:
* A `SettingsController` similar to `ThemeController` that returns a config blob `{ okta_enabled: boolean }` or whatever name/type you think makes sense.
* A `SettingsProvider` that hits that unauthenticated JSON endpoint (unless we already have it in local storage) and provides settings in a context, wrapped here `src/providers/app.tsx`
* Update Login page to look at context instead of `PUBLIC_OKTA_SSO_PATH` var

Reply:
Got it, good to know. I'll replace the env var with a settings endpoint and adjust the login UI. I'll use local store as a cache. In other apps I've used service workers for this kind of cache, maybe something to consider for the future.

Reply:
Ok, cool. If you think service worker is a better mechanism for caching settings, I'm fine with that. I haven't used them much, so it wasn't an intentional choice.